src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -3918,8 +3918,14 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
       }
       def registerDeprecationSuppression(info: AnnotationInfo): Unit =
         if (annotee.isDefined && info.matches(DeprecatedAttr) && !runReporting.suppressionExists(info.pos)) {
-          val (start, end) = rangeFinder()
-          runReporting.addSuppression(Suppression(info.pos, List(MessageFilter.Category(WarningCategory.Deprecation)), start, end, synthetic = true))
+          def suppressAt(start: Int, end: Int): Unit =
+            runReporting.addSuppression(Suppression(info.pos, List(MessageFilter.Category(WarningCategory.Deprecation)), start, end, synthetic = true))
+          rangeFinder() match { case (start, end) => suppressAt(start, end) case _ => }
+          val annsym = annotee.get.symbol
+          if (annsym != null && annsym.isPackageObject && annsym.owner != null && annsym.owner.pos.isDefined) {
+            val p = annsym.owner.pos
+            suppressAt(p.start, p.end)
```

Comment:
this only works if rangepos is enabled

Reply:
please don't crush my dreams.

Reply:
I'll try to come up with a different idea.